controllers/ingress.go

```diff
@@ -51,26 +51,40 @@ func (r *GslbReconciler) ensureIngress(instance *k8gbv1beta1.Gslb, i *v1beta1.In
 		if err != nil {
 			// Creation failed
 			log.Error(err, "Failed to create new Ingress", "Ingress.Namespace", i.Namespace, "Ingress.Name", i.Name)
-			return &reconcile.Result{}, err
+			return err
 		}
 		// Creation was successful
-		return nil, nil
+		return nil
 	} else if err != nil {
 		// Error that isn't due to the service not existing
 		log.Error(err, "Failed to get Ingress")
-		return &reconcile.Result{}, err
+		return err
 	}
 
 	// Update existing object with new spec and annotations
-	found.Spec = i.Spec
-	found.Annotations = i.Annotations
-	err = r.Update(context.TODO(), found)
-
-	if err != nil {
-		// Update failed
-		log.Error(err, "Failed to update Ingress", "Ingress.Namespace", found.Namespace, "Ingress.Name", found.Name)
-		return &reconcile.Result{}, err
+	if !ingressCompare(found, i) {
+		found.Spec = i.Spec
+		found.Annotations = i.Annotations
+		err = r.Update(context.TODO(), found)
+		if errors.IsConflict(err) {
+			r.Log.Info("CONFLICT, Ingress updated by third-party")
```

Comment:
It will still make the actual update, correct? If yes, then the log message is misleading and *looks* like an error

Reply:
Thx for review and good question:

The heartbeat of this pull-request is this:
```go
if !ingressCompare(found, i) {
		found.Spec = i.Spec
		found.Annotations = i.Annotations
		err = r.Update(context.TODO(), found)
		if err != nil {
			// Update failed
			log.Error(err, "Failed to update Ingress", "Ingress.Namespace", found.Namespace, "Ingress.Name", found.Name)
			return err
		}
	}
```
It says, *hey, if you change GSLB annotations, then we revert your changes to the desired state. If you change no GSLB annotations, we don't care.*.

The problem is, between the time that we get and attempt to update a resource, something else has updated the resource, and the version we have is "stale" and it refuses to let you update it, which obviously leads to `r.Update(context.TODO(), found)` returns error -  detected by `errors.IsConflict(err)`.

So in such case we can log info message and return nil. Reconciliation queue will be executed immediately, because and next run would be ok. 

- misleading error: yes I cand fix. What about `re-run reconciliation due to ingress conflict` ?

I just realized, to make it even cleaner we should keep third party annotations on ingress and update only ours if needed:


```go
	if !ingressCompare(found, i) {
		found.Spec = i.Spec
		found.Annotations = repairAnnotations(i.Annotations) // <<< keep original annotations but alter / update our annotations
		err = r.Update(context.TODO(), found)
		if errors.IsConflict(err) {
			r.Log.Info("CONFLICT, Ingress updated by third-party")
			return nil
		}
		if err != nil {
			// Update failed
			log.Error(err, "Failed to update Ingress", "Ingress.Namespace", found.Namespace, "Ingress.Name", found.Name)
			return err
		}
	}
```

Reply:
@kuritka , how about this message:
```golang
r.Log.Info("Ingress has been modified outside of controller, retrying reconciliation", "Ingress.Namespace", found.Namespace, "Ingress.Name", found.Name)
```